### PR TITLE
invalid issuer in UserPrincipalManager (#642)

### PR DESCRIPTION
**Pull request by ekonwlo**
## Summary
Allowing https://login.microsoftonline.com/ issuer to pass the token claim validation

## Issue Type
- Bug fixing

## Starter Names
  - active directory spring boot starter
 

## Additional Information

### CODE REVIEW

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerTest.java [hunk 1]
```diff
@@ -55,6 +59,15 @@ public void invalidIssuer() {
                 .isInstanceOf(BadJWTException.class);
     }
 
+    @Test
+    //TODO: add more generated tokens with other valid issuers to this file. Didn't manage to generate them
```
[the remark this answers is not in the document]
Comment (ekonwlo) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerTest.java]: ok I'll try, but I couldn't figure out how to the token was generated for the test class. I think it was done with the private key from the test resources,

[the remark this answers is not in the document]
Comment (neuqlz) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerTest.java]: Do you mean you don't know how to generate a token for the test case? you can use the private key in the test resources folder to generate it.

[the remark this answers is not in the document]
Comment (ekonwlo) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerTest.java]: yeah, I know, but not sure how to do it, I try with something like this but it fails:

   final byte[] keyBytes = Files.readAllBytes(Paths.get("src/test/resources/test-private-key.txt"));
    final PemObject pem = new PemReader(new InputStreamReader(new ByteArrayInputStream(keyBytes))).readPemObject();
    final PKCS8EncodedKeySpec spec =
            new PKCS8EncodedKeySpec(pem.getContent());
    final KeyFactory kf = KeyFactory.getInstance("RSA");
    final RSAPrivateKey privateKey = (RSAPrivateKey) kf.generatePrivate(spec);

    signer = new RSASSASigner(privateKey);

File: azure-spring-boot/pom.xml [hunk 2]
```diff
@@ -235,6 +235,12 @@
             <artifactId>wiremock-standalone</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>pl.pragmatists</groupId>
+            <artifactId>JUnitParams</artifactId>
+            <version>1.1.1</version>
+            <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (ekonwlo) [reply azure-spring-boot/pom.xml]: Added to a parent pom

**Pull request merged**